auth/httptransport/httptransport.go

```diff
@@ -153,22 +153,21 @@ func NewClient(opts *Options) (*http.Client, error) {
 	if err := opts.validate(); err != nil {
 		return nil, err
 	}
-	// TODO(codyoss): re-add in a future PR
 
-	// tOpts := &transport.Options{
-	// 	Endpoint:           opts.Endpoint,
-	// 	ClientCertProvider: opts.ClientCertProvider,
-	// 	Client:             opts.client(),
-	// }
-	// if io := opts.InternalOptions; io != nil {
-	// 	tOpts.DefaultEndpoint = io.DefaultEndpoint
-	// 	tOpts.DefaultMTLSEndpoint = io.DefaultMTLSEndpoint
-	// }
-	// clientCertProvider, dialTLSContext, err := transport.GetHTTPTransportConfig(tOpts)
-	// if err != nil {
-	// 	return nil, err
-	// }
-	trans, err := newTransport(defaultBaseTransport(nil), opts)
+	tOpts := &transport.Options{
+		Endpoint:           opts.Endpoint,
+		ClientCertProvider: opts.ClientCertProvider,
+		Client:             opts.client(),
+	}
+	if io := opts.InternalOptions; io != nil {
+		tOpts.DefaultEndpoint = io.DefaultEndpoint
+		tOpts.DefaultMTLSEndpoint = io.DefaultMTLSEndpoint
```

Comment:
Is it guaranteed that these properties will be set on `io`? What happens if they aren't and these become empty?

Reply:
It is valid for them to be empty. In those cases it just copies over an empty string.

Reply:
Ok cool, thanks